tests/integration/test_gatekeeper.py

```diff
@@ -0,0 +1,23 @@
+import pytest
+from tests.factory_fixtures.dummy_resource import DummyResource
+from utils.gatekeeper import GatekeeperError
+
+
+@pytest.mark.usefixtures("client_class")
+class TestGatekeeper:
+    def test_empty_set_allows_no_params(self):
+        with pytest.raises(GatekeeperError):
+            self.client.put("/api/dummy", json={"hi": "there"})
```

Comment:
I would like to not have to add logic to the routes file simply for testing if possible. We don't need to test routing so could we bypass it by calling the method directly?

Reply:
That makes sense.  The only thing is, the decorator looks at `request.json.keys()`, so I'll have to figure out how to properly mock the request object and pass it to that method.